dask/dataframe/core.py

```diff
@@ -5176,6 +5181,10 @@ def set_index(
         ["Alice", "Laura", "Ursula", "Zelda"]
         >>> # ^ Now copy-paste this and edit the line above to:
         >>> # ddf2 = ddf.set_index("name", divisions=["Alice", "Laura", "Ursula", "Zelda"])
+
+        Lastly, setting ``sort=False`` will negate the performance impact above
```

Comment:
I think it might be better to move this information to the top. The docstring is quite long, so I expect few users to fully read it and the performance implications of the flag seem quite important.

Maybe we could weave it in with something like: 

>~This realigns the dataset to be sorted by a new column.~ **If ``sort=False``, this function operates exactly like ``pandas.set_index`` and sets the index on the dataframe. If ``sort=True`` (default), this function also sorts the dataframe by the new index**. This can have...
>
>~This function operates exactly like ``pandas.set_index`` except with different performance costs (dask dataframe ``set_index`` is much more expensive). ~ **With ``sort=True``, this function is much more expensive.** Under normal operation this function **then** does an initial pass...

What do you think?

Reply:
Oh ya, that's better. Took most of your suggestion, with only minor adjustment to the last bit. https://github.com/dask/dask/pull/10342/commits/43f79628dce1adca617f477b2c0f37920271490f No terribly strong feelings either way to feel free to push back.

Reply:
This reads well! For context, my idea behind getting rid of 

> This function operates exactly like `pandas.set_index` except with different performance costs

was that the statement seemed wrong to me. By default, `dd.set_index` does not operate exactly like `pandas.set_index`. That's the point behind the `sort` argument you add in this PR. I don't need to have that change to the docstring in your PR though. If you have conflicting thoughts I'd defer this to another issue.

Reply:
@milesgranger: I'ved an update.

cc @jrbourbeau in case you have any thoughts on the docs here. I will merge this later if nothing pops up.